### PR TITLE
Cgi variables

### PR DESCRIPTION
**Pull request by rosamakinen**
just a simple draft, making a map of the environment variables and then transferring the map to a array of strings to be given to execve. Easy to add more env variables when needed. 

cgiEnvironment**["REQUEST_URI"]** The interpreted pathname of the requested document or CGI (relative to the document root)

cgiEnvironment**["GATEWAY_INTERFACE"]** The revision of the Common Gateway Interface that the server uses. (pretty much always CGI/1.1)

cgiEnvironment**["SERVER_PROTOCOL"]**  The name and revision of the information protocol the request came in with. (HTTP_VERSION aka HTTP/1.1)

cgiEnvironment**["REQUEST_METHOD"]** POST or GET

cgiEnvironment**["SCRIPT_FILENAME"]** is The full pathname of the current CGI

### CODE REVIEW

File: src/CgiHandler.cpp [hunk 1]
```diff
@@ -1,15 +1,56 @@
...
-int executeCgi()
+static char **transferToString(std::map<std::string, std::string> cgiEnvironment)
+{
+	char **environmentString = NULL;
+	environmentString = new char*[cgiEnvironment.size() + 1]; 
+	int i = 0;
+
+	for (std::map<std::string, std::string>::iterator it = cgiEnvironment.begin(); it != cgiEnvironment.end(); ++it)
+	{
+		environmentString[i] = strdup((it->first + "=" + it->second).c_str());
```
[the remark this answers is not in the document]
Comment (rosamakinen) [reply src/CgiHandler.cpp]: done

File: src/CgiHandler.cpp [hunk 2]
```diff
@@ -1,15 +1,56 @@
...
+	// cgiEnvironment["PATH_INFO"] = request.getUri(); if needed
+	// other env variables if needed
+
+	return cgiEnvironment;
+}
+
 
 static int	executeChild()
 {
 	// int result = execve(path_to_command(bin/python-command), command_with_flags(from-te-cgi-bin-directory), environment_variables);
```
[the remark this answers is not in the document]
Comment (rosamakinen) [reply src/CgiHandler.cpp:20]: donetski

File: src/CgiHandler.cpp [hunk 3]
```diff
@@ -1,15 +1,56 @@
 
 #include "../include/CgiHandler.hpp"
+#include "CgiHandler.hpp"
+
```
[the remark this answers is not in the document]
Comment (rosamakinen) [reply src/CgiHandler.cpp:3]: yesss

File: include/HttpRequestHandler.hpp [hunk 4]
```diff
@@ -7,8 +7,10 @@
 #include "FileHandler.hpp"
 #include "ExceptionManager.hpp"
 #include "Client.hpp"
+#include "CgiHandler.hpp"
 #include "Server.hpp"
 
+class CgiHandler;
```
[the remark this answers is not in the document]
Comment (rosamakinen) [reply include/HttpRequestHandler.hpp]: yes, whopsies, fixed now

File: src/CgiHandler.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (merituulie) [reply src/CgiHandler.cpp:20]: did you push this uncmmented part?

File: src/CgiHandler.cpp [hunk 5]
```diff
@@ -1,15 +1,52 @@
...
+static std::map<std::string, std::string> initCgiEnvironment(HttpRequest request)
+{
+	std::map<std::string, std::string> cgiEnvironment;
+
+	cgiEnvironment["REQUEST_URI"] = request.getUri();
+	cgiEnvironment["GATEWAY_INTERFACE"] = GATEWAY_VERSION;
+	cgiEnvironment["SERVER_PROTOCOL"] = HTTP_VERSION;
+	cgiEnvironment["REQUEST_METHOD"] = request.translateMethod(request.getMethod());
```
Comment (merituulie) [inline src/CgiHandler.cpp:11]: nice, really nice!

File: include/WebServer.hpp [hunk 6]
```diff
@@ -4,6 +4,7 @@
 #define MESSAGE_BUFFER		1048576
 
 #define HTTP_VERSION		"HTTP/1.1"
+#define GATEWAY_VERSION		"CGI/1.1"
```
Comment (merituulie) [inline include/WebServer.hpp:7]: greatness, look at you go!

Comment (rosamakinen) [reply include/WebServer.hpp:7]: i know right!

File: src/CgiHandler.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (merituulie) [reply src/CgiHandler.cpp:20]: @rosamakinen

**Review by merituulie — APPROVED** [review]

**Review by forbidden-arts — APPROVED** [review]

**Pull request merged**